apps/comps/rpc/middleware/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { StandardRPCJsonSerializer } from "@orpc/client/standard";
+import {
+  Context,
+  ErrorMap,
+  Meta,
+  Middleware,
+  ORPCErrorConstructorMap,
+  os,
+} from "@orpc/server";
+import { unstable_cache } from "next/cache";
+import type { Logger } from "pino";
+
+export function cacheMiddleware<
+  TInContext extends Context & { logger: Logger },
+  TInput,
+  TOutput,
+  TErrorMap extends ErrorMap,
+  TMeta extends Meta,
+>(options?: {
+  key?: string[];
+  revalidate?: number;
```

Comment:
can we call this `revalidateSecs` so its clear what the unit is?

Really I'd love for us to start using a real Duration type throughout the case base for stuff like this to really eliminate the risk of bugs related to misused units for durations